### PR TITLE
Add Aliyun SLB support

### PR DESCRIPTION
**Pull request by niusmallnan**
It is very similar to ELB.

### CODE REVIEW

**Comment by prachidamle** [comment]
@niusmallnan can you split the commits into two - one for the vendor/trash dependencies and other for code changes - easier to go through that way

**Comment by niusmallnan** [comment]
@prachidamle Sure, have done it.

**Comment by yasker** [comment]
@prachidamle 

Can you help to review the updated version?

**Comment by nayang777** [comment]
rancher/rancher#8921

**Comment by prachidamle** [comment]
@niusmallnan LGTM - just a minor comment below, but since this looks very similar to ELB, @janeczku can you also review?

File: providers/aliyunslb/slb.go [hunk 1]
```diff
@@ -0,0 +1,478 @@
...
+	"github.com/rancher/external-lb/providers"
+)
+
+const (
+	ProviderName = "Aliyun SLB"
+	ProviderSlug = "aliyun_slb"
+)
+
+const (
+	TagResoueceFlag    = "external-lb/rancher"
```
Comment (prachidamle) [inline providers/aliyunslb/slb.go]: nit pick: seems like a typo here.

File: providers/aliyunslb/slb.go [hunk 2]
```diff
@@ -0,0 +1,478 @@
...
+}
+
+func (p *AliyunSLBProvider) GetName() string {
+	return ProviderName
+}
+
+func (p *AliyunSLBProvider) HealthCheck() error {
+	_, err := p.slbClient.DescribeRegions()
+	if err != nil {
+		return fmt.Errorf("Falied to list SLB available regions: %v", err)
```
Comment (janeczku) [inline providers/aliyunslb/slb.go]: Typo: "_Falied_ to list"

**Comment by janeczku** [comment]
See comment about typo. Otherwise LGTM 👍

**Comment by yasker** [comment]
@niusmallnan Can you make PR to community catalog as well?

**Pull request merged**